packages/dm-core/src/hooks/useRecipe.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { useBlueprint, TUiRecipe } from '../index'
+
+const findRecipe = (
+  initialUiRecipe: any,
+  recipeName: string | null,
+  recipes: any[]
+) => {
+  // If recipe is defined, find and return the ui recipe from available recipes.
+  if (recipeName) {
+    return recipes.find((recipe: any) => recipe.name == recipeName)
+  }
+  // If not recipe is defined, use initialize recipe, or the first from recipes list or lastly fallback.
+  // TODO: how to check that is empty?
+  if (initialUiRecipe) {
+    return initialUiRecipe
+  }
+  if (recipes.length > 0) {
+    return recipes[0]
+  }
+  return {
+    name: 'yaml',
+    plugin: 'yaml',
+    config: {},
+  }
+}
+
+interface IUseRecipe {
+  recipe: TUiRecipe
+  isLoading: boolean
+  error: Error | null
+}
+
+/**
+ * Hook to get a recipe
+ *
+ * @docs Hooks
+ *
+ * @usage
+ * Code example:
+ * ```
+ * import { useRecipe } from '@data-modelling-tool/core'
+ *
+ * const { recipe, isLoading, error} = useRecipe('someType', 'someRecipe')
+ * if (isLoading) return <div>Loading...</div>
+ * if (error) return <div>Error getting the recipe</div>
+ *
+ * if (recipe) {
+ *   ...
+ * }
+ * ```
+ *
+ * @param typeRef - The reference to the blueprint to retrieve a recipe for
+ * @param recipeName - Name of recipe to find (optional)
+ * @returns A list containing the blueprint document, a boolean representing the loading state, and an Error, if any.
+ */
+export const useRecipe = (
+  typeRef: string,
+  recipeName: string | null
```

Comment:
This way, you would have to send in `recipeName == null ` if you would want to use the default one. Write instead 
```suggestion
  recipeName?: string
```
, then you can skip the input variable if yoou don't want to specify